### PR TITLE
feat(medusa): Use query relation load strategy on Carts

### PR DESCRIPTION
**Pull request by olivermrbl**

### CODE REVIEW

File: packages/medusa/src/repositories/cart.ts [hunk 1]
```diff
@@ -1,5 +1,18 @@
-import { Cart } from "../models"
+import { ExtendedFindConfig } from "@medusajs/types"
 import { dataSource } from "../loaders/database"
+import { Cart } from "../models"
+
+export const CartRepository = dataSource.getRepository(Cart).extend({
+  async findOne(options: ExtendedFindConfig<Cart>) {
+    const options_ = { ...options }
+
+    const [cart] = await this.find(options_)
```
Comment (adrien2p) [inline packages/medusa/src/repositories/cart.ts]: suggestion: No need to copy the options as they are not mutated.

File: packages/medusa/src/repositories/cart.ts [hunk 2]
```diff
@@ -1,5 +1,18 @@
...
+
+    if (cart?.payment_sessions) {
```
Comment (adrien2p) [inline packages/medusa/src/repositories/cart.ts]: suggestion: `?.length`

**Review by adrien2p — APPROVED** [review]
LGTM

**Comment by adrien2p** [comment]
> LGTM

Just need the CI to pass

**Comment by olivermrbl** [comment]
Can I merge this @adrien2p ?

**Review by adrien2p — APPROVED** [review]
Lgtm, one question though, should we also do the same for the find? Otherwise it might break someone listing the carts right?

**Comment by olivermrbl** [comment]
@adrien2p – ~~we don't support listing carts, so it could only happen if someone extended their setup with that functionality.~~ Sorry, we do at the service level but not in the API.

Still, it wont break but just be less performant as it would use the join strategy. Think it's fine no?

**Comment by olivermrbl** [comment]
In any case, we could do it in a follow-up iteration

**Comment by adrien2p** [comment]
> @adrien2p – ~we don't support listing carts, so it could only happen if someone extended their setup with that functionality.~ Sorry, we do at the service level but not in the API.
> 
> Still, it wont break but just be less performant as it would use the join strategy. Think it's fine no?

@olivermrbl We can do it in a followup pr but now if someone use the list cart then the payment sessions won’t be available on the carts.

**Pull request merged**